src/Ruleset.php

```diff
@@ -1470,13 +1472,30 @@ public function populateTokenListeners()
 
             $tokenizers = [];
             $vars       = get_class_vars($sniffClass);
-            if (isset($vars['supportedTokenizers']) === true) {
+            if (empty($vars['supportedTokenizers']) === false) {
```

Comment:
As this is a deprecation not a feature removal, we shouldn't be changing the behaviour. With the `isset()` check, it looks like an empty array would mean that a sniff would never run, whereas now that we're checking for `empty()`, that empty array value gets rewritten to `PHP` (in the `else` clause here).

Reply:
Good point - I made the change to [avoid a deprecation notice being thrown for the property being set to an empty array](https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/894/files#diff-2b3b004ba20442d720b6b4f5a2e9040987b9d049f8a2dfb3a406d580060075e7R54) - which is not terribly useful, but not something problematic either. This is tested via the `ListensForEmptySniff` test case.

But you're right, this does change the behaviour, so maybe the deprecation notice should be wrapped in its own conditions, independently of the conditions registering the tokenizers for the sniff.

What do you think ?

Reply:
I've update the PR with the above. Let me know what you think.